pyproject.toml

```diff
@@ -10,36 +10,29 @@
 [build-system]
 build-backend = 'mesonpy'
 requires = [
-    "meson-python>=0.12.1",
+    "meson-python>=0.13.1",
     "Cython>=0.29.35",  # when updating version, also update check in meson.build
     "pybind11>=2.10.4",
     "pythran>=0.12.0",
-    # `wheel` is needed for non-isolated builds, given that `meson-python`
-    # doesn't list it as a runtime requirement (at least in 0.5.0)
-    "wheel",
 
-    # NumPy dependencies - to update these, sync from
-    # https://github.com/scipy/oldest-supported-numpy/, and then
-    # update minimum version to match our install_requires min version
-    # ----------------------------------------------------------------
-
-    # now matches minimum supported version, keep here as separate requirement
-    # to be able to sync more easily with oldest-supported-numpy
+    # When numpy 2.0.0rc1 comes out, we should update this to build against 2.0,
+    # and then runtime depend on the range 1.22.X to <2.3. No need to switch to
+    # 1.25.2 in the meantime (1.25.x is the first version which exports older C
+    # API versions by default).
 
     # default numpy requirements
     "numpy==1.22.4; python_version<='3.10' and platform_python_implementation != 'PyPy'",
     "numpy==1.23.2; python_version=='3.11' and platform_python_implementation != 'PyPy'",
-
-    # For Python versions which aren't yet officially supported,
-    # we specify an unpinned NumPy which allows source distributions
-    # to be used and allows wheels to be used as soon as they
-    # become available.
-    "numpy; python_version>='3.12'",
+    # For Python versions which aren't yet officially supported, we specify an
+    # unpinned NumPy which allows source distributions to be used and allows
+    # wheels to be used as soon as they become available.
+    "numpy>='1.26.0b1'; python_version>='3.12'",
```

Comment:
Believe it or not, it looks like `pypa/packaging` is choking on the single quotes around the beta version number for some reason. It seems to have nothing to do with the beta specification though. This seems to fix the failure shown in CI with `pypa/build` when I try locally:

```diff
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -26,7 +26,7 @@ requires = [
     # For Python versions which aren't yet officially supported, we specify an
     # unpinned NumPy which allows source distributions to be used and allows
     # wheels to be used as soon as they become available.
-    "numpy>='1.26.0b1'; python_version>='3.12'",
+    "numpy>=1.26.0b1; python_version>='3.12'",
     "numpy; python_version>='3.9' and platform_python_implementation=='PyPy'",
 ]

```

Reply:
Always fun to guess which things fail with quotes, and which things without quotes. Let's see if it's happy now ....